### PR TITLE
CNS-418: changed all LavaError to LavaFormatError/Warning/panic

### PR DESCRIPTION
**Pull request by oren-lava**

### CODE REVIEW

File: common/fixation_entry.go [hunk 1]
```diff
@@ -116,8 +115,9 @@ func (fs *FixationStore) AppendEntry(
 ) error {
 	safeIndex, err := types.SanitizeIndex(index)
 	if err != nil {
-		details := map[string]string{"index": index}
-		return utils.LavaError(ctx, ctx.Logger(), "AppendEntry_invalid_index", details, "invalid non-ascii entry")
+		return utils.LavaFormatError("invalid non-ascii entry", err,
```
Comment (Yaroms) [inline common/fixation_entry.go:118]: description string is a duplicate in all of the functions, change it to give us an idea where it came from

File: common/fixation_entry.go [hunk 2]
```diff
@@ -283,8 +282,9 @@ func (fs *FixationStore) getUnmarshaledEntryForBlock(ctx sdk.Context, safeIndex
 func (fs *FixationStore) FindEntry(ctx sdk.Context, index string, block uint64, entryData codec.ProtoMarshaler) bool {
 	safeIndex, err := types.SanitizeIndex(index)
 	if err != nil {
-		details := map[string]string{"index": index}
-		utils.LavaError(ctx, ctx.Logger(), "FindEntry_invalid_index", details, "invalid non-ascii entry")
+		utils.LavaFormatError("invalid non-ascii entry", err,
```
Comment (Yaroms) [inline common/fixation_entry.go:285]: description string is a duplicate in all of the functions, change it to give us an idea where it came from

File: common/fixation_entry.go [hunk 3]
```diff
@@ -301,8 +301,9 @@ func (fs *FixationStore) FindEntry(ctx sdk.Context, index string, block uint64,
 func (fs *FixationStore) GetEntry(ctx sdk.Context, index string, entryData codec.ProtoMarshaler) bool {
 	safeIndex, err := types.SanitizeIndex(index)
 	if err != nil {
-		details := map[string]string{"index": index}
-		utils.LavaError(ctx, ctx.Logger(), "GetEntry_invalid_index", details, "invalid non-ascii entry")
+		utils.LavaFormatError("invalid non-ascii entry", err,
```
Comment (Yaroms) [inline common/fixation_entry.go:304]: description string is a duplicate in all of the functions, change it to give us an idea where it came from

File: common/fixation_entry_index.go [hunk 4]
```diff
@@ -67,8 +67,9 @@ func (fs FixationStore) GetAllEntryIndices(ctx sdk.Context) []string {
 func (fs *FixationStore) getEntryVersionsFilter(ctx sdk.Context, index string, block uint64, filter func(*types.Entry) bool) (blocks []uint64) {
 	safeIndex, err := types.SanitizeIndex(index)
 	if err != nil {
-		details := map[string]string{"index": index}
-		utils.LavaError(ctx, ctx.Logger(), "getEntryVersionsFilter", details, "invalid non-ascii entry")
+		utils.LavaFormatError("invalid non-ascii entry", err,
```
Comment (Yaroms) [inline common/fixation_entry_index.go:70]: description string is a duplicate in all of the functions, change it to give us an idea where it came from

File: x/conflict/keeper/msg_server_detection.go [hunk 5]
```diff
@@ -21,22 +21,30 @@ func (k msgServer) Detection(goCtx context.Context, msg *types.MsgDetection) (*t
 	logger := k.Keeper.Logger(ctx)
 	clientAddr, err := sdk.AccAddressFromBech32(msg.Creator)
 	if err != nil {
-		return nil, utils.LavaError(ctx, logger, "conflict_detection", map[string]string{"client": msg.Creator, "error": err.Error()}, "parsing client address")
+		return nil, utils.LavaFormatWarning("parsing client address", err,
```
Comment (Yaroms) [inline x/conflict/keeper/msg_server_detection.go:24]: change desc - invalid client address

File: x/conflict/keeper/msg_server_detection.go [hunk 6]
```diff
@@ -21,22 +21,30 @@ func (k msgServer) Detection(goCtx context.Context, msg *types.MsgDetection) (*t
...
+			utils.Attribute{Key: "client", Value: msg.Creator},
+		)
 	}
 	if msg.FinalizationConflict != nil && msg.ResponseConflict == nil && msg.SameProviderConflict == nil {
 		err := k.Keeper.ValidateFinalizationConflict(ctx, msg.FinalizationConflict, clientAddr)
 		if err != nil {
-			return nil, utils.LavaError(ctx, logger, "Finalization_conflict_detection", map[string]string{"client": msg.Creator, "error": err.Error()}, "Simulation: finalization conflict detection error")
+			return nil, utils.LavaFormatWarning("Simulation: finalization conflict detection error", err,
```
Comment (Yaroms) [inline x/conflict/keeper/msg_server_detection.go:31]: Simulation: invalid finalization conflict detection

File: x/conflict/keeper/msg_server_detection.go [hunk 7]
```diff
@@ -21,22 +21,30 @@ func (k msgServer) Detection(goCtx context.Context, msg *types.MsgDetection) (*t
...
+				utils.Attribute{Key: "client", Value: msg.Creator},
+			)
 		}
 	} else if msg.FinalizationConflict == nil && msg.ResponseConflict == nil && msg.SameProviderConflict != nil {
 		err := k.Keeper.ValidateSameProviderConflict(ctx, msg.SameProviderConflict, clientAddr)
 		if err != nil {
-			return nil, utils.LavaError(ctx, logger, "same_provider_conflict_detection", map[string]string{"client": msg.Creator, "error": err.Error()}, "Simulation: same provider conflict detection error")
+			return nil, utils.LavaFormatWarning("Simulation: same provider conflict detection error", err,
```
Comment (Yaroms) [inline x/conflict/keeper/msg_server_detection.go:38]: desc: Simulation: invalid same provider conflict detection

File: x/conflict/keeper/msg_server_detection.go [hunk 8]
```diff
@@ -21,22 +21,30 @@ func (k msgServer) Detection(goCtx context.Context, msg *types.MsgDetection) (*t
...
+				utils.Attribute{Key: "client", Value: msg.Creator},
+			)
 		}
 	} else if msg.FinalizationConflict == nil && msg.ResponseConflict != nil && msg.SameProviderConflict == nil {
 		err := k.Keeper.ValidateResponseConflict(ctx, msg.ResponseConflict, clientAddr)
 		if err != nil {
-			return nil, utils.LavaError(ctx, logger, "response_conflict_detection", map[string]string{"client": msg.Creator, "error": err.Error()}, "Simulation: response conflict detection error")
+			return nil, utils.LavaFormatWarning("Simulation: response conflict detection error", err,
```
Comment (Yaroms) [inline x/conflict/keeper/msg_server_detection.go:45]: description: Simulation: invalid response conflict detection

File: common/fixation_entry.go (hunk 3 above)
Comment (oren-lava) [reply common/fixation_entry.go:304]: done

File: common/fixation_entry_index.go (hunk 4 above)
Comment (oren-lava) [reply common/fixation_entry_index.go:70]: done

File: x/conflict/keeper/msg_server_detection.go (hunk 6 above)
Comment (oren-lava) [reply x/conflict/keeper/msg_server_detection.go:31]: done

File: x/conflict/keeper/msg_server_detection.go (hunk 8 above)
Comment (oren-lava) [reply x/conflict/keeper/msg_server_detection.go:45]: done

File: x/subscription/keeper/subscription.go [hunk 9]
```diff
@@ -169,32 +164,31 @@ func (k Keeper) CreateSubscription(
 		// new subscription needs a default project
 		err = k.projectsKeeper.CreateAdminProject(ctx, consumer, plan)
 		if err != nil {
-			details := map[string]string{
-				"err": err.Error(),
-			}
-			return utils.LavaError(ctx, logger, "CreateSubscription", details, "failed to create default project")
+			return utils.LavaFormatError("failed to create default project", err)
```
Comment (Yaroms) [inline x/subscription/keeper/subscription.go:167]: should be a warning, inner functions should decide on warning or error for different scenarious

**Review by Yaroms — CHANGES_REQUESTED** [review]

File: x/epochstorage/keeper/fixated_params.go [hunk 10]
```diff
@@ -161,9 +166,14 @@ func (k Keeper) GetFixatedParamsForBlock(ctx sdk.Context, fixationKey string, bl
...
 			if block < earliestEpochStart {
-				err = utils.LavaError(ctx, k.Logger(ctx), "fixated_params_too_early", map[string]string{"error": "tried to read for block that is earlier than earliest_epoch_start", "block": strconv.FormatUint(block, 10), "earliest": strconv.FormatUint(earliestEpochStart, 10)}, "invalid block requested, that is lower than earliest block in memory")
+				err = utils.LavaFormatError("invalid block requested, that is lower than earliest block in memory", fmt.Errorf("tried to read for block that is earlier than earliest_epoch_start"),
+					utils.Attribute{Key: "block", Value: block},
+					utils.Attribute{Key: "earliest", Value: earliestEpochStart},
+				)
 			} else {
-				err = utils.LavaError(ctx, k.Logger(ctx), "fixated_params_for_block_empty", map[string]string{"error": "tried to read index: " + thisIdxKey + " but wasn't found", "block": strconv.FormatUint(block, 10)}, "invalid block requested, that is lower than saved fixation memory")
+				err = utils.LavaFormatError("invalid block requested, that is lower than saved fixation memory", fmt.Errorf("tried to read index: "+thisIdxKey+" but wasn't found"),
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/epochstorage/keeper/fixated_params.go:174]: done

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 11]
```diff
@@ -201,14 +213,14 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 			burnSucceeded, err2 := k.BurnClientStake(ctx, relay.SpecId, clientAddr, burnAmount, false)
 
 			if err2 != nil {
-				details["amountToBurn"] = burnAmount.String()
-				details["error"] = err2.Error()
-				return errorLogAndFormat("relay_payment_burn", details, "BurnUserStake failed on user")
+				return nil, utils.LavaFormatWarning("BurnUserStake failed on user", err2,
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go:216]: done

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 12]
```diff
@@ -201,14 +213,14 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
...
+					utils.Attribute{Key: "amountToBurn", Value: burnAmount},
+				)
 			}
 			if !burnSucceeded {
-				details["amountToBurn"] = burnAmount.String()
-				details["error"] = "insufficient funds or didn't find user"
-				return errorLogAndFormat("relay_payment_burn", details, "BurnUserStake failed on user, did not find user, or insufficient funds")
+				return nil, utils.LavaFormatWarning("BurnUserStake failed on user, did not find user, or insufficient funds", fmt.Errorf("insufficient funds or didn't find user"),
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go:221]: done

File: x/pairing/keeper/msg_server_stake_client_test.go [hunk 13]
```diff
@@ -28,8 +28,8 @@ func TestNewStakeClient(t *testing.T) {
 		stake sdk.Coin
 		valid bool
 	}{
-		{"MinStake", sdk.NewCoin(spec.MinStakeClient.Denom, spec.MinStakeClient.Amount.Sub(sdk.NewInt(1))), false},
-		{"InsufficientFunds", sdk.NewCoin(epochstoragetypes.TokenDenom, sdk.NewInt(amount+1)), false},
+		{"InsufficientFunds", sdk.NewCoin(spec.MinStakeClient.Denom, spec.MinStakeClient.Amount.Sub(sdk.NewInt(1))), false},
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/pairing/keeper/msg_server_stake_client_test.go:31]: done

File: x/subscription/keeper/subscription.go [hunk 14]
```diff
@@ -107,33 +108,27 @@ func (k Keeper) CreateSubscription(
...
 	}
 
 	creatorAcct, err := sdk.AccAddressFromBech32(creator)
 	if err != nil {
-		details := map[string]string{
-			"creator": creator,
-			"error":   err.Error(),
-		}
-		return utils.LavaError(ctx, logger, "CreateSubscription", details, "invalid creator")
+		return utils.LavaFormatWarning("cannot create subcscription with invalid creator", err,
```
[the remark this answers is not in the document]
Comment (oren-lava) [reply x/subscription/keeper/subscription.go:121]: done

File: x/subscription/keeper/subscription.go (hunk 9 above)
Comment (oren-lava) [reply x/subscription/keeper/subscription.go:167]: done

**Pull request merged**